bootstrapgenerator/src/main/java/nl/openweb/hippo/groovy/XmlGenerator.java

```diff
@@ -69,29 +62,37 @@ private XmlGenerator() {
     /**
      * Parse file to updater node
      *
-     *
      * @param sourceDir the directory to read the resources from
-     * @param script the script to use for source
+     * @param script    the script to use for source
      * @return Node object representing the groovy updater to marshall to xml
      */
     public static Node getUpdateScriptNode(final File sourceDir, final ScriptClass script) {
         final Updater updater = script.getUpdater();
         final Node rootnode = XmlGenerator.createNode(updater.name());
         final List<Object> properties = rootnode.getNodeOrProperty();
-        properties.add(createProperty(JCR_PRIMARY_TYPE, HIPPOSYS_UPDATERINFO, ValueType.NAME));
-        properties.add(createProperty(HIPPOSYS_BATCHSIZE, updater.batchSize(), ValueType.LONG));
-        addStringPropertyIfNotEmpty(properties, HIPPOSYS_DESCRIPTION, updater.description());
-        properties.add(createProperty(HIPPOSYS_DRYRUN, updater.dryRun(), ValueType.BOOLEAN));
-        addStringPropertyIfNotEmpty(properties, HIPPOSYS_PARAMETERS, getValueOrFileContent(script, sourceDir, updater.parameters()));
-        if(StringUtils.isBlank(updater.xpath())) {
-            addStringPropertyIfNotEmpty(properties, HIPPOSYS_PATH, updater.path());
+        final Map<String, Object> propertiesForUpdater = PropertyCollector.getPropertiesForUpdater(script, sourceDir);
+        for (Map.Entry<String, Object> entry : propertiesForUpdater.entrySet()) {
+            String valueType = getValueType(entry);
+            Object value = entry.getValue();
+            if (HIPPOSYS_SCRIPT.equals(entry.getKey())) {
+                value = wrap(value.toString());
+            }
+            properties.add(createProperty(entry.getKey(), value, valueType));
         }
-        addStringPropertyIfNotEmpty(properties, HIPPOSYS_QUERY, updater.xpath());
-        addStringPropertyIfNotEmpty(properties, HIPPOSYS_SCRIPT, wrap(script.getContent()));
-        properties.add(createProperty(HIPPOSYS_THROTTLE, updater.throttle(), ValueType.LONG));
         return rootnode;
     }
 
+    private static String getValueType(final Map.Entry<String, Object> entry) {
```

Comment:
This value will never be an Array?

Reply:
No, the property can only be of one type.